### ISSUE TITLE
Games preview always gets stuck on 47888/47889

### ISSUE THREAD

**Issue by ankx22** [mtkennerly/ludusavi#389]
### Ludusavi version

v0.25.0

### Operating system

Windows

### Installation method

Standalone

### Description

Whenever I starts Ludosavi and run the preview games, the bar always goes quickly to 47888/47889 and then just remains there permenantly. Never completes and so I am never able to backup anything. Is it something wrong that I am doing?
![Screenshot_1](https://github.com/user-attachments/assets/61958fac-7231-4ac3-8bab-2c922d5154e6)

**Comment by mtkennerly** [comment]
Hi! There's probably one particular game that has a bad save location in the manifest, causing Ludusavi to scan too much.

Could you please open Command Prompt and try running this? (replace `C:\path\to\ludusavi.exe` with the actual path on your system)

```
set RUST_LOG=ludusavi=trace
C:\path\to\ludusavi.exe backup --preview
```

Once it gets stuck, please upload a copy of `%APPDATA%\ludusavi\ludusavi_rCURRENT.log` here. The log should show which game it's stuck on.

**Comment by ankx22** [comment]
[ludusavi_rCURRENT.log](https://github.com/user-attachments/files/16920133/ludusavi_rCURRENT.log)

Heres the log. Is the problem with MSFS?

**Comment by mtkennerly** [comment]
Yeah, that looks like the issue, since the files for that game can be pretty huge. It's come up a few other times (https://github.com/mtkennerly/ludusavi/issues/243, https://github.com/mtkennerly/ludusavi/issues/375).

On the "other" screen, in the backup exclusions list, you can add a file entry for `~/AppData/Local/Packages/Microsoft/FlightSimulator_8wekyb3d8bbwe` to exclude those files from the scan entirely.

Long-term, Ludusavi should probably be able to report individual files for a game instead of waiting for the whole game to finish scanning, or at least be able to skip a game if it hits certain size/time limits.

**Comment by epicyeeto** [comment]
I would like to add to this that minecraft bedrock edition also does this it look likes it tries and scans everything like textures, behavior packs etc.,

**Comment by mtkennerly** [comment]
You can now click on the progress bar to see exactly which game is causing the problem and how long it's been scanning :)

> <img width="1282" height="999" alt="Image" src="https://github.com/user-attachments/assets/7c63395b-01f5-4bd4-9f86-1854c0aad2ec" />

Once you confirm the problematic game, you could make an empty custom game set to "override" so that Ludusavi doesn't have anything to scan for it (or you could just remove some specific paths from the entry):

> <img width="1239" height="279" alt="Image" src="https://github.com/user-attachments/assets/e3e3ed3e-82c3-4348-9c6d-6727f40661a7" />

Also, once you do that and you've done one full, successful scan of all games, then you can just scan one game at a time, even if you close and reopen Ludusavi:

> https://github.com/user-attachments/assets/56aeb04e-f3aa-4c4c-9848-482dd137518e

**Comment by codedpalette** [comment]
I'm having the same problem. In my case the problematic game is "Heroes of Might and Magic III"

Here is the [ludusavi_debug.log](https://github.com/user-attachments/files/28080544/ludusavi_debug.log) from running `flatpak run com.github.mtkennerly.ludusavi --debug backup --preview "Heroes of Might and Magic III"`. From what I understand it correctly finds it in Lutris installed games, but then for some reason scans my whole `$HOME/Games` folder, even though I'm running it on Linux, not Windows

The proposed solution with custom games works partially, as I'm able to backup all my other games. However, running individual scan for Heroes of Might and Magic III still hangs indefinitely and cannot be cancelled from the ui

**Comment by mtkennerly** [comment]
@codedpalette Thanks for the log file! In your case, it looks like the issue has to do with how Ludusavi checks non-Steam shortcuts in Steam:

```
[2026-05-21T00:59:28.633Z] TRACE [ludusavi::scan::steam] Found Steam shortcut: app_name='Heroes of Might and Magic III', official_title='Heroes of Might and Magic III', id=3955765141, start_dir='"/home/daniel"'
```

Ludusavi assumes that the shortcut's start directory is where the game is installed. One of the paths for Heroes of Might and Magic III is `<base>/Games`, which means to look for a `Games` folder inside of the install folder. Since the shortcut starts in `/home/daniel`, it starts scanning the entire `/home/daniel/Games` 😅 

I thought the default directory for a Steam shortcut is usually the same folder with the executable. Did you set `/home/daniel` intentionally, or was that the default?

**Comment by codedpalette** [comment]
I've added HoMM through Lutris and didn't change the defaults. I checked some other Lutris installed shortcuts, and they all point to `home/daniel`, but the save backup works. And `<base>/Games` path for HoMM is windows only, so it shouldn't try to scan it on linux, no?

**Comment by mtkennerly** [comment]
Your Lutris config is fine. The issue is coming from your Steam config, where you have a non-Steam shortcut for Heroes of Might and Magic III. When Ludusavi scans your Steam root, it also scans shortcuts added to Steam.

> And `<base>/Games` path for HoMM is windows only, so it shouldn't try to scan it on linux, no?

Ludusavi generally ignores OS constraints when checking paths (unless it's something that can't exist, like `<winAppData>` on Linux without Wine).

**Comment by codedpalette** [comment]
I've removed everything from directory path in steam shortcut and the sync works now, thanks
